distributed/client.py

```diff
@@ -1027,7 +1024,7 @@ async def _start(self, timeout=no_default, **kwargs):
                     asynchronous=self._asynchronous,
                     **self._startup_kwargs,
                 )
-            except (OSError, socket.error) as e:
+            except OSError as e:
```

Comment:
Are we sure this should be dropped?

Reply:
This should be okay as `IOError`, `EnvironmentError`, `socket.error`, and a few other error types were all unified in Python 3.3 

```python
Python 3.9.2 | packaged by conda-forge | (default, Feb 21 2021, 05:02:20)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.22.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import socket

In [2]: OSError is socket.error
Out[2]: True
```

I think this type of simplification is what `pyupgrade` helps with